evproc/processor.py

```diff
@@ -308,11 +311,16 @@ def process(self, ev):
                   ``StopProcessing`` exception raised by a processor.
         """
 
-        # Walk through all the processors and process the event
-        for proc in self._get_procs(ev):
-            try:
-                proc(ev)
-            except exc.StopProcessing as ex:
-                return ex.retval
+        event_deque = collections.deque([ev])
+        while event_deque:
+            ev = event_deque.popleft()
+            # Walk through all the processors and process the event
+            for proc in self._get_procs(ev):
+                try:
+                    additional_events = proc(ev)
+                except exc.StopProcessing as ex:
+                    return ex.retval
```

Comment:
Yeah, I knew about the default value for ``retval``, and was going to mention the idiom I usually use for that sort of thing (``_unset = object()`` and use ``_unset`` as the default), but I wanted to see what sort of defense you could come up with :)

I think you have a good point that the behavior may be unchanged, but don't forget that this is pluggable; you could have some event processors for one event that are coded to the old version, while others for that same event are coded to the new version.  Still, the new ones could probably be written to deal with that case, but it should probably be called out in the documentation…